### DISCUSSION TITLE
Trying to make a time dependent chest item checker and mover

### DISCUSSION THREAD

**Discussion by dumbpinoob** [cc-tweaked/CC-Tweaked#1792] [Q&A]
I'm completely new to lua so I have no idea how to start exactly. I am trying to make a program that will print contents of a chest onto a page at a certain ingame time and then move all items out of the chest (but only after counting and printing) to another. I don't want the entire code though, I'd only a list of commands that I need it to run so i can figure the details out myself. If I have further questions I will come back and ask.
I thank you in advance for helping me out.

**Comment by Lupus590** [comment]
It sounds like you'll be helped a lot by the documentation website: https://tweaked.cc/

**Reply by SquidDev** [reply]
More specifically, you'll want [the `list()` function](https://tweaked.cc/generic_peripheral/inventory.html#v:list) to scan an inventory, and then [`pushItems()`](https://tweaked.cc/generic_peripheral/inventory.html#v:pushItems) to move those items to another chest.

**Reply by dumbpinoob** [reply]
Probably should have been more specific about what I need. With the help of the wiki and some basic lua tutorials I have now figured out how to make a code that checks what day it is and then lists all items if the day is different than the last execution. What I can't figure out now is how to make it loop indefinitely and how to make it print the item list on a page.

**Reply by Lupus590** [reply]
A `while true do` loop will loop forever as `true` is always true.
`print` is what most people use to write to the screen, there is also `write` and a few other functions in the term API.
`textutils.tabulate` (I might have that name wrong) might be useful too.

**Reply by dumbpinoob** [reply]
![image](https://github.com/cc-tweaked/CC-Tweaked/assets/115364080/94fa83a1-556d-4f77-ba93-07511570b4d4)
![image](https://github.com/cc-tweaked/CC-Tweaked/assets/115364080/97e167e3-1538-4412-9805-6d946c315564)
I'm completely stuck on this. I got everything else I need except for getting ingame day. For some reason it always starts on the real day of the month and then counts up with each passing ingame day. I want it to count only ingame days.

**Reply by SquidDev** [reply]
`local` should be used when you need to declare a variable (in which case it should be of the form `local my_variable = some_expression`). All those other lines starting with `local` are very much confusing the parser — that code should look something like:

```lua
local printer = peripheral.find("printer")
printer.newPage()
printer.write(day)
printer.setCursorPos(1, 3)
```

**Reply by dumbpinoob** [reply]
Still getting the same error after removing all the locals. Looks like this now:
![image](https://github.com/cc-tweaked/CC-Tweaked/assets/115364080/30b1c99b-d78e-4e0f-ace4-0ed8a11841fb)

**Reply by SquidDev** [reply]
The first of those is still a little wrong. You need to specify the variable, so it should be `local printer = peripheral.find("printer")`.

**Reply by dumbpinoob** [reply]
I lied, I don't have this figured out. The printer prints now but doesn't change lines automatically. I have no idea how I'd go about setting that up.
